packages/grid/x-data-grid-pro/src/hooks/features/treeData/useGridTreeDataPreProcessors.tsx

```diff
@@ -212,6 +219,12 @@ export const useGridTreeDataPreProcessors = (
   );
   useGridRegisterStrategyProcessor(privateApiRef, TREE_DATA_STRATEGY, 'filtering', filterRows);
   useGridRegisterStrategyProcessor(privateApiRef, TREE_DATA_STRATEGY, 'sorting', sortRows);
+  useGridRegisterStrategyProcessor(
+    privateApiRef,
+    TREE_DATA_STRATEGY,
+    'visibleRows',
```

Comment:
**Optional:** To make it a gerund like `rowTreeCreation`, `filtering`, and `sorting`. (Doesn't sound great though 😄 )
```suggestion
    'visibleRowsComputation',
```

Reply:
Renamed to `visibleRowsLookupCreation` 👍